### DISCUSSION TITLE
Versioning at different levels/layers

### DISCUSSION THREAD

**Discussion by stephanbcbauer** [eclipse-tractusx/eclipse-tractusx.github.io#181] [Ideas]
**Current situation**

Currently, there is no versioning of the documentation on the website directly. According to the issue #113, this is planned -> I think it's good. But I think that it is not enough to have the versioning on this level. Since the individual KITS can be available in different versions, I think versioning is needed on another level, or in addition.

For example, as a user of the Connector Kit, I would like to see the documented version of 0.1.6 and 0.3.0. This is currently not possible. And I think several KITS have the UseCase

**Proposal**

For me, plugin [docs-multi-instance](https://docusaurus.io/docs/docs-multi-instance) seems to be exactly what we require? The example, a mobile APP is mentioned here with different versions in the documentation of IOS SDK and Android SDK. I think it is definitely worth verifying the plug-in.

@danielmiehle @maximilianong @stefan-ettl

**Comment by santiagoejm** [comment]
Hi @stephanbcbauer ! Thanks for opening the discussion, this is the main subject that is been lately focused on. I know Luis is working on alternatives to implement the versioning, because as you mentioned the multi-instance plugin was one of the options and I got the chance to test it locally, however there are some conclusions that are leading us in other direction. 

Among the reasons for example, as the documentation grows the building time will exponentially do as well, the Docusaurus documentation warns about the size of each doc-instance and whether if it is proper to use this feature (makes me think that the documentation of our project, if not now very soon will be big, how this would affect the plugin's performance?). Other factor (although quite dumb and relatively "fixable") is that by default all the dropdown menus in the NavBar that contains the version's options of each instance are displayed at the same time regardless of the kit documentation we are viewing at the moment, this issue was raised some while ago and closed with no resolved status, however there is an "unofficial" solution by one of the member of the community, you can check it in here [https://github.com/facebook/docusaurus/issues/3930](https://github.com/facebook/docusaurus/issues/3930) (it still works), but should we consider the possibility that in the future a Docusaurus upgrade would cause a major inconvenient/impediment for us to upgrade because it will require a major change in our structure if this is not fixed? We are already blocked to update the current Docusaurus version to the latest because the open-api-plugin is not yet compatible with it. 

Just thinking out loud some of the "technicals" things that make me be cautious about this plugin for our project, there are other "process related" things that have been discussed and would be great to share those with you, maybe in a meeting? :)

**Comment by stephanbcbauer** [comment]
@santiagoejm For me, the current versioning concept is totally fine and works very well. So I will close the discussion. Thank you.
